src/main/java/com/brandwatch/internship/fetchandstoretweets/services/QueryService.java

```diff
@@ -0,0 +1,82 @@
+package com.brandwatch.internship.fetchandstoretweets.services;
+
+import com.brandwatch.internship.fetchandstoretweets.entities.Query;
+import com.brandwatch.internship.fetchandstoretweets.repositories.QueryRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.repository.query.ParameterOutOfBoundsException;
+import org.springframework.stereotype.Service;
+
+import javax.inject.Singleton;
+import java.util.Comparator;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicLong;
+
+@Service
+@Singleton
+public class QueryService {
+
+    private AtomicLong topId;
+    private QueryRepository queryRepository;
+
+    @Autowired
+    private QueryService(QueryRepository queryRepository) {
+
+        try {
+            this.topId = new AtomicLong(queryRepository.findFirstByOrderByIdDesc().getId());
+        }
+        catch (ParameterOutOfBoundsException e) {
+            e.printStackTrace();
+            this.topId = new AtomicLong();
+        }
+
+        this.queryRepository = queryRepository;
+    }
+
+    public List<Query> getAllQueries() {
+        List<Query> queries = queryRepository.findAll();
+        queries.sort(Comparator.comparing(Query::getId));
+
+        return queries;
+    }
+
+    public Query createQuery(String searchString) {
+        if (searchString.equals("")) {
+            return emptyQuery();
+        }
+        Query query = new Query(topId.incrementAndGet(), searchString);
+
+        queryRepository.save(query);
```

Comment:
Does the save method return the object that's been stored? Because if it does this method could be a one-liner imo:

```java
public Query createQuery(String searchString) {
return queryRepository.save(new Query(topId.incrementAndGet(), searchString););
}
```

A two-liner is also fine of course if you find that more readable.

Reply:
You are right. I thought its void but it returns the saved object. I will modify it.